### PR TITLE
Introduce demo prover

### LINKED ISSUE #251: Release demo-prover
Currently, we have a private repo where we've built out the prover logic. We should migrate the code to the mono-repo, and rename to `prover-template` or `risc0-prover-template`.

### PR DESCRIPTION
**Pull request by citizen-stig**
# Major changes

* Adds `demo-prover`
* Adds new crate `examples/const-rollup-config` which holds compile time parameters, shared between `demo-rollup` and `demo-prover`

Resolves #251

### CODE REVIEW

File: examples/demo-prover/host/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        100000000,
+        sequencer_private_key.default_address(),
+        vec![
+            99, 101, 108, 101, 115, 116, 105, 97, 49, 122, 102, 118, 114, 114, 102, 97, 113, 57,
+            117, 100, 54, 103, 57, 116, 52, 107, 122, 109, 115, 108, 112, 102, 50, 52, 121, 115,
+            97, 120, 113, 102, 110, 122, 101, 101, 53, 119, 57,
+        ],
+        &sequencer_private_key,
+        &sequencer_private_key,
+    );
```
Comment (preston-evans98) [inline examples/demo-prover/host/src/main.rs:44]: nit: Can we update this to match the demo-rollup? That uses some constants to make the purpose of these magic values more clear

Comment (citizen-stig) [reply examples/demo-prover/host/src/main.rs]: Done!

File: examples/demo-prover/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   `cargo build`. The downloaded code will be stored in a directory called "riscv-guest-shim", which should be placed in the
+   `methods/guest` directory.
+2. Build `.rlib` binaries from the modified stdlib. To build the standard library, use the command below (replacing the path).
+   To minimize extraneous code, you may want to comment out the dependencies section of `Cargo.toml` and the body of
+   `rollup.rs` before continuing.
+
+```
+__CARGO_TESTS_ONLY_SRC_ROOT="/path/to/riscv-guest-shim/rust-std" cargo run -Zbuild-std=core,alloc,proc_macro,panic_abort,std -Zbuild-std-features=compiler-builtins-mem --test --release
+```
+
```
[the remark this answers is not in the document]
Comment (citizen-stig) [reply examples/demo-prover/README.md]: Moved to the separate document

**Comment by citizen-stig** [comment]
> What do you think?

Aggree, please checkout new README

**Review by bkolad — APPROVED** [review]

File: adapters/celestia/Cargo.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ tendermint = "0.32"
 tendermint-proto = "0.32"
 
 # Convenience
-anyhow = "1.0.62"
+anyhow = "1.0.68"
```
Comment (preston-evans98) [inline adapters/celestia/Cargo.toml:16]: Is there a `workspace` version of anyhow we can use instead?

**Review by preston-evans98 — APPROVED** [review]

File: adapters/celestia/Cargo.toml (hunk 3 above)
Comment (citizen-stig) [reply adapters/celestia/Cargo.toml:16]: Good point!

**Pull request merged**